x-pack/plugins/security/server/lib/authorization/mode.test.js

```diff
@@ -6,30 +6,76 @@
 
 import { authorizationModeFactory } from './mode';
 
-const createMockXpackInfoFeature = (allowRbac) => {
-  return {
-    getLicenseCheckResults() {
-      return {
-        allowRbac
-      };
-    }
-  };
-};
-
-describe(`#useRbac`, () => {
+class MockXPackInfoFeature {
+  constructor(allowRbac) {
+    this.allowRbac = allowRbac;
+  }
+
+  mockSetAllowRbac(allowRbac) {
+    this.allowRbac = allowRbac;
+  }
+
+  getLicenseCheckResults() {
+    return {
+      allowRbac: this.allowRbac
+    };
+  }
+}
+
+describe(`#useRbacForRequest`, () => {
+  test(`throws an Error if request isn't specified`, async () => {
+    const mockXpackInfoFeature = new MockXPackInfoFeature(false);
+    const mode = authorizationModeFactory(mockXpackInfoFeature);
+
+    expect(() => mode.useRbacForRequest()).toThrowErrorMatchingInlineSnapshot(
+      `"Invalid value used as weak map key"`
+    );
+  });
+
+  test(`throws an Error if request is "null"`, async () => {
+    const mockXpackInfoFeature = new MockXPackInfoFeature(false);
+    const mode = authorizationModeFactory(mockXpackInfoFeature);
+
+    expect(() => mode.useRbacForRequest(null)).toThrowErrorMatchingInlineSnapshot(
+      `"Invalid value used as weak map key"`
+    );
+  });
+
   test(`returns false if xpackInfoFeature.getLicenseCheckResults().allowRbac is false`, async () => {
-    const mockXpackInfoFeature = createMockXpackInfoFeature(false);
+    const mockXpackInfoFeature = new MockXPackInfoFeature(false);
     const mode = authorizationModeFactory(mockXpackInfoFeature);
+    const request = {};
```

Comment:
__optional nit:__ there is also `import { requestFixture } from '../__tests__/__fixtures__/request';` created for cases like this, but not a huge deal.

Reply:
Nice! I didn't know that, I'll use this instead